typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -181,7 +183,15 @@ export abstract class HyperlaneDeployer<
         );
       }
 
-      const currentIsm = await connectionClient.interchainSecurityModule();
+      let currentIsm = await connectionClient.interchainSecurityModule();
+      // in case the above returns zero address, fetch the defaultISM from the mailbox
+      if (currentIsm === ethers.constants.AddressZero) {
+        const mailbox: Mailbox = Mailbox__factory.connect(
+          config.mailbox,
+          connectionClient.signer,
+        );
+        currentIsm = await mailbox.defaultIsm();
+      }
 
       // set interchain security module if not already set (and configured)
       let configuredIsm;
```

Comment:
configuredIsm will never be defined outside of `(config.interchainSecurityModule)`